new-log-viewer/src/contexts/UrlContextProvider.tsx

```diff
@@ -0,0 +1,209 @@
+import React, {
+    createContext,
+    useEffect,
+    useState,
+} from "react";
+import {
+    SEARCH_PARAM_NAME,
+    HASH_PARAM_NAME,
+    UrlSearchParams,
+    UrlHashParams,
+    UrlSearchParamUpdatesType,
+    UrlHashParamUpdatesType,
+    UrlParamsType
+} from "../typings/url";
+
+
+const UrlContext = createContext <UrlParamsType>({} as UrlParamsType);
+
+interface UrlContextProviderProps {
+    children: React.ReactNode
+}
+
+let searchParams = new URLSearchParams(window.location.search.substring(1));
```

Comment:
As per our offline discussion, isn't this supposed to be a Map / Object of parameters that is already parsed?